functional-tests/src/test/java/org/infinispan/online/service/endpoint/HotRodTester.java

```diff
@@ -34,20 +36,130 @@ public void testPutPerformance(URL urlToService, long timeout, TimeUnit timeUnit
+      assertNull(stringCache.get(key));
+
+      stringCache.put(key, "value");
+      assertEquals("value", stringCache.get(key));
+
+      stringCache.put(key, "newValue");
+      assertEquals("newValue", stringCache.get(key));
+
+      stringCache.remove(key);
+      assertNull(stringCache.get(key));
+
+      RemoteCache<String, Integer> intCache = cachingService.getCache();
+
+      Integer intValue = 5;
+      intCache.put(key, intValue);
+      assertEquals(intValue, intCache.get(key));
+
+      RemoteCache<String, Long> longCache = cachingService.getCache();
+
+      Long longValue = 25L;
+      longCache.put(key, longValue);
+      assertEquals(longValue, longCache.get(key));
+
+      RemoteCache<String, Short> shortCache = cachingService.getCache();
+
+      Short shortValue = 22;
+      shortCache.put(key, shortValue);
+      assertEquals(shortValue, shortCache.get(key));
+
+      RemoteCache<String, Byte> byteCache = cachingService.getCache();
+
+      Byte byteValue = 15;
+      byteCache.put(key, byteValue);
+      assertEquals(byteValue, byteCache.get(key));
+
+      RemoteCache<String, Character> charCache = cachingService.getCache();
+
+      Character charValue = 'a';
+      charCache.put(key, charValue);
+      assertEquals(charValue, charCache.get(key));
+
+      RemoteCache<String, Double> doubleCache = cachingService.getCache();
+
+      Double doubleValue = 25.5d;
+      doubleCache.put(key, doubleValue);
+      assertEquals(doubleValue, doubleCache.get(key));
+
+      RemoteCache<String, Float> floatCache = cachingService.getCache();
+
+      Float floatValue = 27.5f;
+      floatCache.put(key, floatValue);
+      assertEquals(floatValue, floatCache.get(key));
+
+      RemoteCache<String, Boolean> boolCache = cachingService.getCache();
+
+      boolCache.put(key, true);
+      assertEquals(true, boolCache.get(key));
+
+      boolCache.put(key, false);
+      assertEquals(false, boolCache.get(key));
+
+      RemoteCache<String, byte[]> byteArrayCache = cachingService.getCache();
+      byte[] byteArrayValue = generateConstBytes(4096);
+      byteArrayCache.put(key, byteArrayValue);
+      assertArrayEquals(byteArrayValue, byteArrayCache.get(key));
+   }
+
+   public void evictionTest(URL hotRodService) {
+      RemoteCacheManager cachingService = getRemoteCacheManager(hotRodService);
+      RemoteCache<String, byte[]> byteArrayCache = cachingService.getCache();
+      byte[] firstValue = generateConstBytes(4096);
+
+      String key = "key";
+      byteArrayCache.put(key, firstValue);
+      assertArrayEquals(firstValue, byteArrayCache.get(key));
+
+      // when the first entry is evicted, the test ends
+      long counter = 0;
+      while (byteArrayCache.get(key) != null) {
+         byte[] value = generateConstBytes(4096);
+
+         byteArrayCache.put(key + counter++, value);
+         assertArrayEquals(value, byteArrayCache.get(key));
+      }
+   }
+
+   public static byte[] generateConstBytes(int size) {
```

Comment:
I guess this one deserves to be extracted into some util class. We might want to generate random bytes in other tests. See `TestObjectCreator`.